api/src/main/java/org/apache/iceberg/actions/ExpireSnapshots.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+import org.apache.iceberg.Snapshot;
+
+/**
+ * An action that expires snapshots in a table.
+ * <p>
+ * Similar to {@link org.apache.iceberg.ExpireSnapshots} but may use a query engine to distribute
+ * parts of the work.
+ */
+public interface ExpireSnapshots extends Action<ExpireSnapshots, ExpireSnapshots.Result> {
```

Comment:
I don't have `<ThisT>` in `ExpireSnapshots` as we don't expect engine-specific interfaces.